openghg/standardise/_standardise.py

```diff
@@ -489,72 +508,48 @@ def standardise_flux(
-#     Returns:
-#         None
-#     """
-#     from gzip import compress
-#     import tempfile
-#     from openghg.objectstore import PAR
-#     from openghg.client import get_function_url, get_auth_key
-
-#     auth_key = get_auth_key()
-#     fn_url = get_function_url(fn_name="get_par")
-#     # First we need to get a PAR to write the data
-
-#     response = _post(url=fn_url, auth_key=auth_key)
-#     par_json = response.content
-
-#     par = PAR.from_json(json_str=par_json)
-#     # Get the URL to upload data to
-#     par_url = par.uri
-
-#     if filepath is not None and data is None:
-#         filepath = Path(filepath)
-#         MB = 1e6
-#         file_size = Path("somefile.txt").stat().st_size / MB
-
-#         mem_limit = 50  # MiB
-#         if file_size < mem_limit:
-#             # Read the file, compress it and send the data
-#             file_data = filepath.read_bytes()
-#             compressed_data = compress(data=file_data)
-#         else:
-#             tmp_dir = tempfile.TemporaryDirectory()
-#             compressed_filepath = Path(tmp_dir.name).joinpath(f"{filepath.name}.tar.gz")
-#             # Compress in place and then upload
-#             with tarfile.open(compressed_filepath, mode="w:gz") as tar:
-#                 tar.add(filepath)
-
-#             compressed_data = compressed_filepath.read_bytes()
-#     elif data is not None and filepath is None:
-#         compressed_data = gzip.compress(data)
-#     else:
-#         raise ValueError("Either filepath or data must be passed.")
-
-#     # Write the data to the object store
-#     put_response = _put(url=par_url, data=compressed_data, auth_key=auth_key)
-
-#     print(str(put_response))
+        if bucket is None:
+            bucket = get_writable_bucket(name=store)
+        return standardise(
+            data_type="emissions",
+            bucket=bucket,
+            filepath=filepath,
+            species=species,
+            source=source,
+            domain=domain,
+            database=database,
+            database_version=database_version,
+            model=model,
+            high_time_resolution=high_time_resolution,
+            period=period,
+            continuous=continuous,
+            chunks=chunks,
+            overwrite=overwrite,
+        )
+
+
+def standardise_from_binary_data(
+    bucket: str, data_type: str, binary_data: bytes, metadata: dict, file_metadata: dict, **kwargs: Any
+) -> Optional[dict]:
+    dclass = get_data_class(data_type)
+    with dclass(bucket) as dc:
+        result = dc.read_data(
+            binary_data=binary_data, metadata=metadata, file_metadata=file_metadata, **kwargs
+        )
+    return result
+
+
+def standardise_from_remote_source(bucket: str, data_type: str, data: dict, **kwargs: Any) -> Optional[dict]:
+    dclass = get_data_class(data_type)
+    with dclass(bucket) as dc:
+        result = dc.store_data(data=data, **kwargs)
+    return result
+
+
+def standardise_via_transform(
```

Comment:
See Issue #588 in relation to the plans for the transform functionality. This is meant to be expanded and sit in parallel to standardise rather than going through the same interface.

Reply:
I added a _transform.py module to the top level of openghg.transform and put `transform_emissions_data` in there, which does the same thing that this function did. (I added a doc string as well.)